monkey/infection_monkey/payload/ransomware/ransomware.py

```diff
@@ -70,12 +73,9 @@ def _send_telemetry(self, filepath: Path, success: bool, error: str):
         encryption_attempt = FileEncryptionTelem(str(filepath), success, error)
         self._telemetry_messenger.send_telemetry(encryption_attempt)
 
-    def _leave_readme_in_target_directory(self, interrupt: threading.Event):
-        if interrupt.is_set():
-            logger.debug("Received a stop signal, skipping leave readme")
-            return
-
+    @interruptible_function(msg="Received a stop signal, skipping leave readme")
```

Comment:
Not sure it's worth interrupting. This operation is fast and explaining what happened to the directory is worth it I think.

Reply:
The drive we're writing to could be very slow, or a network share. The resource be overutilized and very slow to respond. We have no way to control how long this operation takes.

If/when we add a force kill, maybe this becomes moot.